core/src/forward_renderer.cpp

```diff
@@ -166,9 +246,167 @@ void ForwardRenderer::color_pass() {
     }
 }
 
-void ForwardRenderer::skybox_pass() {}
-void ForwardRenderer::transparent_pass() {}
-void ForwardRenderer::post_process_pass() {}
+void ForwardRenderer::gui_pass(uint16_t idx) {
+    using namespace components;
+    auto sceneOpt = Scene::get_active_scene();
+    if (!sceneOpt) {
+        return;
+    }
+
+    Scene& scene = sceneOpt.value();
+    entt::registry& registry = scene.get_registry();
+    mat4 invProj;
+    glm::mat4 view;
+    auto windowSize = m_engine.get_window_module().lock()->get_window_size();
+    //=CAMERA===========================
+    auto cameras = registry.view<Transform, EditorCamera, Name>();
+
+    for (auto& cam : cameras) {
+        auto goOpt = scene.get_game_object_from_handle(cam);
+        if (!goOpt) {
+            continue;
+        }
+
+        GameObject go = goOpt.value();
+        Transform& transform = go.get_component<Transform>();
+        EditorCamera& editorCamera = go.get_component<EditorCamera>();
+        Name& name = go.get_component<Name>();
+
+        const glm::vec3 pos = transform.get_position();
+        const glm::vec3 lookTarget = editorCamera.get_look_target();
+        const glm::vec3 up = editorCamera.get_up();
+
+        const float fovY = editorCamera.get_fov();
+        const float aspectRatio = float((float)windowSize.x / (float)windowSize.y);
+        const float zNear = editorCamera.get_z_near();
+        const float zFar = editorCamera.get_z_far();
+
+        // Set view and projection matrix for view 0.
+        {
+            view = glm::lookAt(pos, lookTarget, up);
+            glm::mat4 proj = glm::perspective(fovY, aspectRatio, zNear, zFar);
+            bgfx::setViewTransform(idx, &view[0][0], &proj[0][0]);
+        }
+    }
+
+    auto entities = registry.view<Transform, InstanceMesh, Material, Name, PostProcessing>();
```

Comment:
So all GUI entities will need postprocessing components?

Reply:
this is some redundant example code, anything that happens inside of the "gui_pass" using the "idx" provided by the function will be drawn to the GUI Framebuffer.

the framebuffer is then provided alongside the (colour / geometry) pass as a texture for the post processing stack where anything in the GUI pass that has an alpha > 0 will be rendered over the top of the (colour / geometry) texture

Reply:
This makes sense thank you!